querybook/webapp/ui/Tag/HoverIconTag.tsx

```diff
@@ -4,28 +4,73 @@ import React from 'react';
 import { Icon } from 'ui/Icon/Icon';
 import type { AllLucideIconNames } from 'ui/Icon/LucideIcons';
 
-import { ITagProps, Tag } from './Tag';
+import { ITagProps, Tag, TagGroup } from './Tag';
 
-export interface IHoverIconTagProps extends ITagProps {
+export interface IHoverIconTagProps extends Omit<ITagProps, 'children'> {
+    tagName: string;
+    tagType?: string;
+    tagIcon?: string;
     iconOnHover?: AllLucideIconNames;
     onIconHoverClick?: (e?: React.MouseEvent) => any;
 }
 export const HoverIconTag = React.forwardRef<
     HTMLSpanElement,
     IHoverIconTagProps
->(({ iconOnHover, onIconHoverClick, children, ...tagProps }, ref) => {
-    const hoverDOM = iconOnHover ? (
-        <div className="HoverIconTag-hover" onClick={onIconHoverClick}>
-            <Icon name={iconOnHover} />
-        </div>
-    ) : null;
-
-    const className = clsx(tagProps['className'], 'HoverIconTag');
-
-    return (
-        <Tag {...tagProps} ref={ref} className={className}>
-            {children}
-            {hoverDOM}
-        </Tag>
-    );
-});
+>(
+    (
+        {
+            tagName,
+            tagType,
+            tagIcon,
+            iconOnHover,
+            onIconHoverClick,
+            ...tagProps
+        },
+        ref
+    ) => {
+        const hoverDOM = iconOnHover ? (
+            <div className="HoverIconTag-hover" onClick={onIconHoverClick}>
+                <Icon name={iconOnHover} />
+            </div>
+        ) : null;
+
+        const className = clsx(tagProps['className'], 'HoverIconTag');
+
+        const { tooltip, tooltipPos, color, mini, onClick } = tagProps;
+
+        const iconDOM = tagIcon && (
+            <Icon name={tagIcon as any} size={16} className="mr4" />
+        );
+
+        if (tagType) {
+            return (
+                <span
+                    aria-label={tooltip}
+                    data-balloon-pos={tooltipPos}
+                    onClick={onClick}
+                    ref={ref}
+                    className={className}
+                >
+                    <TagGroup>
```

Comment:
i just realized TagGroup is just a div wrapper, can you convert it into styled components and pass in the props above to there instead?